chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/renderingmiddlewares/attachmentMiddleware.tsx

```diff
@@ -77,6 +80,69 @@ const Attachment = (props: any) => {
     );
 };
 
+const Spinner = (props: any) => {
+    const spinnerStyle: any = {
+        boxSizing: "border-box",
+        borderRadius: "50%",
+        borderWidth: props.borderWidth || 2,
+        borderStyle: "solid",
+        borderColor: "rgb(0,120,212) rgb(199,224,244) rgb(199,224,244)",
+        borderImage: "initial",
+        animationName: "spin",
+        animationDuration: "1.3s",
+        animationIterationCount: "infinite",
+        animationTimingFunction: "cubic-bezier(0.53, 0.21, 0.29, 0.67)",
+        width: props.size || 20,
+        height: props.size || 20,
+    };
+
+    return (
+        <>
+            <style>{`
+                @keyframes spin {
+                    0% {
+                        transform: rotate(0deg);
+                    }
+                    100% {
+                        transform: rotate(360deg);
+                    }
+                }
+            `}</style>
+            <div className="spinner" style={spinnerStyle}></div>
+        </>
+    );
+};
+
+const ScanInProgressAttachment = (props: any) => {
+    const renderer = () => (
+        <div style={{display: "flex", padding: "10px 10px 10px 8px", width: "100%"}}>
+            <div style={{fontSize: 12, fontFamily: "Segoe UI, Arial, sans-serif"}}> {props.textCard.attachment.name} </div>
+            <div style={{marginLeft: "auto", paddingRight: "10px"}}>
+                <Spinner size={16}/>
+            </div>
+        </div>
+    );
+
+    return (
+        <Attachment {...props} iconData={FileScanInProgressIcon} imageCard={undefined} renderer={renderer}/>
+    );
+};
+
+const MaliciousAttachment = (props: any) => {
+    const renderer = () => (
+        <div style={{display: "flex", padding: "10px 10px 10px 8px", width: "100%"}}>
+            <div style={{fontSize: 12, fontFamily: "Segoe UI, Arial, sans-serif"}}> {props.textCard.attachment.name} </div>
+            <div style={{marginLeft: "auto", paddingRight: "10px"}}>
+                <img src={CrossIcon} alt="Malicious" />
+            </div>
+        </div>
+    );
```

Comment:
it seems like the only difference is to use the spinner or the "cross" image, could we refctor to have a function that based on params returns the proper element?, this will eliminate the need of duplicated code